packages/itwinui-react/src/core/FileUpload/FileUploadTemplate.tsx

```diff
@@ -32,6 +33,22 @@ export type FileUploadTemplateProps = {
    * @default 'or drag & drop it here.'
    */
   subtitle?: React.ReactNode;
+  /**
+   * Files selected. Used for default uploaded file output only when one file is selected.
+   */
+  files?: Array<File>;
```

Comment:
Seems like all these new props are targeting single file upload. What happens in a future PRs when we'd add same styles for multiple file upload? Add more props or change these ones (breaking change)?

Reply:
I have changed it so that users can only pass in a file into `FileUploadTemplate` if `accessMultiple` is `false`. Once we have a default GUI for multiple files, we can change `files` (which is now called `data` with my recent changes) to be an array of files only when `accessMultiple` is `true`. This should allow us to implement a standard GUI for multiple file uploads down the road without causing any breaking changes

Reply:
Maybe it makes sense to move 
```
data?: File;
dataLabel?: string;
dataDescription?: string;
dataSvg?: JSX.Element;
```
to new type. Then users could provide single file with metadata or array of files with their metadata. 

Right now, even if you change it to accept Files[], how would user pass label, description and svg for each file?
Although this PR is targeting single file upload only, you should think how it would work with multiple files to not have breaking changes in the future.

Reply:
> The two screenshots for the react visual tests look the same to me. Is one supposed to be without a file having been uploaded?
> 

Yes! Good catch. This has been fixed

Reply:
These props have been moved to its own type. Now when we support multiple file uploads, users can pass an array of `DataProps` which feature their files along with the option to put whatever label, description, and svg they choose for each file

Reply:
What do you think about renaming it to `FileDataProps`? A bit more specific than just DataProps.

Also, now that it's moved, inner props name, eg. "dataLabel", seems a bit redundant. It can be just `label` as it is "data label" from design (child of DataProps).

Reply:
These names have been changed!